support-frontend/app/services/PayPalNvpService.scala

```diff
@@ -90,14 +92,48 @@ class PayPalNvpService(apiConfig: PayPalConfig, wsClient: WSClient) extends Touc
       "RETURNURL" -> returnUrl,
       "CANCELURL" -> cancelUrl,
       "BILLINGTYPE" -> "MerchantInitiatedBilling",
-      "NOSHIPPING" -> "1"
+      "NOSHIPPING" -> noShipping
     )
 
     nvpRequest(paymentParams).map { resp =>
       retrieveNVPParam(resp, "TOKEN")
     }
   }
 
+  // Sets up a payment by contacting PayPal and also fetches the users details.
```

Comment:
I vaguely remember we were using a very out of date paypal something, was it the library here or something else?

Reply:
It is the framework that is out of date not a specific library. Unfortunately we can't upgrade because Zuora doesn't support any other method of integration (as far as I'm aware, we could re investigate this I suppose)

Reply:
got it, that's useful information 👍